src/AllenAlgebra.elm

```diff
@@ -0,0 +1,72 @@
+module AllenAlgebra exposing (..)
+
+{-| The relations defined by an allen algebra.
+
+<https://www.wikiwand.com/en/Allen's_interval_algebra#/Relations>
+
+-}
+
+
+type AllenRelation
+    = Overlaps
+    | OverlapsInverse
+    | During
+    | DuringInverse
+    | Meets
+    | MeetsInverse
+    | Starts
+    | StartsInverse
+    | Finishes
+    | FinishesInverse
+    | Before
+    | After
+    | Equal
+
+
+relation : { r | start : Int, end : Int } -> { p | start : Int, end : Int } -> AllenRelation
```

Comment:
I feel like there's a fuzz we could write here that relates to `Order`.

If the records are degenerates (points instead of ranges), `relation` ought to behave like `compare`:

```elm
Fuzz.fuzz2 Fuzz.int Fuzz.int "" (\x y -> case compare x y of
  LT -> Expect.equal Before (relation {start = x, end = x} {start = y, end = y})
  EQ -> Expect.equal Equal (relation {start = x, end = x} {start = y, end = y})
  GT -> Expect.equal After (relation {start = x, end = x} {start = y, end = y})
)
```

That seem right?

Also, it seems like we don't really need `Int` here, can could get away with just knowing that the values are `comparable`.

Reply:
Awesome, that test found some issues with the algebra. Added the test and fixed the issues. Thanks!